sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -80,13 +83,49 @@ case class CreateTableLikeCommand(
         s"Source table in CREATE TABLE LIKE cannot be temporary: '$sourceTable'")
     }
 
-    val tableToCreate = catalog.getTableMetadata(sourceTable).copy(
-      identifier = targetTable,
-      tableType = CatalogTableType.MANAGED,
-      createTime = System.currentTimeMillis,
-      lastAccessTime = -1).withNewStorage(locationUri = None)
+    val sourceTableDesc = catalog.getTableMetadata(sourceTable)
 
-    catalog.createTable(tableToCreate, ifNotExists)
+    sourceTableDesc.tableType match {
+      case CatalogTableType.MANAGED | CatalogTableType.EXTERNAL | CatalogTableType.VIEW => // OK
+      case o => throw new AnalysisException(
+        s"CREATE TABLE LIKE is not allowed when the source table is ${o.name}")
+    }
+
+    if (DDLUtils.isDatasourceTable(sourceTableDesc) &&
+        sourceTableDesc.tableType != CatalogTableType.MANAGED) {
+      throw new AnalysisException(
+        "CREATE TABLE LIKE is not allowed when the source table is an external table created " +
```

Comment:
After rethinking it, it should be allowed. We just need to recreate the file path under the database directory.